ratatui-widgets/src/table.rs

```diff
@@ -1676,15 +1864,63 @@ mod tests {
         fn length_constraint() {
             // without selection, more than needed width
             let table = Table::default().widths([Length(4), Length(4)]);
-            assert_eq!(table.get_column_widths(20, 0, 0), [(0, 4), (5, 4)]);
+            assert_eq!(
+                table.get_column_widths(20, 0, 0),
+                [
+                    Rect {
+                        x: 0,
+                        width: 4,
+                        y: 0,
+                        height: 1
+                    },
```

Comment:
Long amounts of setup / tests tend to obscure the meat of the tests in a way that makes it tedious to confirm correctness.

One of the big things to do that makes tests easy to confirm as correct is minimize the vertical height of things like this in a way that makes sense:

```suggestion
                    Rect::new(0, 4, 0, 1),
```

etc.